src/pineko/fonll.py

```diff
@@ -0,0 +1,281 @@
+
+    # save final "fonll" fktable
+    fk_folder = Path(configs.load(configs.detect(cfg))["paths"]["fktables"]) / str(
+        theoryid
+    )
+    fk_folder.mkdir(exist_ok=True)
+    output_path_fk = fk_folder / fonll_info.dataset_name
+    combined_fk.write_lz4(output_path_fk)
+
+
+FNS_BASE_PTO = {"FONLL-A": 1, "FONLL-B": 1, "FONLL-C": 2, "FONLL-D": 2, "FONLL-E": 3}
+MIXED_ORDER_FNS = ["FONLL-B", "FONLL-D"]
+
+# Mixed FONLL schemes
+MIXED_FNS_LIST = [
+    "FONLL-FFNS",
+    "FONLL-FFN0",
+    "FONLL-FFNS",
+    "FONLL-FFNS",
+    "FONLL-FFN0",
+    "FONLL-FFNS",
+    "FONLL-FFNS",
+]
+MIXED_NFFF_LIST = [3, 3, 4, 4, 4, 5, 5]
+MIXED_PARTS_LIST = [
+    "full",
+    "full",
+    "massless",
+    "massive",
+    "full",
+    "massless",
+    "massive",
+]
+
+# plain FONLL schemes
+FNS_LIST = [
+    "FONLL-FFNS",
+    "FONLL-FFN0",
+    "FONLL-FFNS",
+    "FONLL-FFN0",
+    "FONLL-FFNS",
+]
+NFFF_LIST = [3, 3, 4, 4, 5]
+PARTS_LIST = ["full" for _ in range(5)]
+
+
+def produce_ptos(fns, is_mixed_or_damp):
+    """Produce the list of PTOs needed for the requested fns."""
+    base_pto = FNS_BASE_PTO[fns]
+    # mixed FONLL
+    if fns in MIXED_ORDER_FNS:
+        return [
+            base_pto + 1,
+            base_pto + 1,
+            base_pto,
+            base_pto + 1,
+            base_pto + 1,
+            base_pto,
+            base_pto + 1,
+        ]
+    # plain FONLL with damping
+    elif is_mixed_or_damp:
+        return [base_pto for _ in range(7)]
+    # plain FONLL without damping
+    else:
+        return [base_pto for _ in range(5)]
+
+
+def produce_fonll_recipe(fonll_fns, damp):
+    """Produce the different theory cards according to which FONLL is asked for."""
+    fonll_recipe = []
+    is_mixed_or_damp = fonll_fns in MIXED_ORDER_FNS or damp != 0
+    fns_list = MIXED_FNS_LIST if is_mixed_or_damp else FNS_LIST
+    nfff_list = MIXED_NFFF_LIST if is_mixed_or_damp else NFFF_LIST
+    parts_list = MIXED_PARTS_LIST if is_mixed_or_damp else PARTS_LIST
+    for fns, nfff, po, part in zip(
+        fns_list, nfff_list, produce_ptos(fonll_fns, is_mixed_or_damp), parts_list
+    ):
+        fonll_recipe.append(
+            {
+                "FNS": fns,
+                "NfFF": nfff,
+                "PTO": po,
+                "FONLLParts": part,
+            }
+        )
+        # In a mixed FONLL scheme we only subract the resummed terms that are
+        # present in the FFNS scheme at nf+1. E.g. for FONLL-B in FFN03 we
+        # only subract up to NLL since there is no NNLL in FFNS4
+        if fonll_fns in MIXED_ORDER_FNS and fns == "FONLL-FFN0":
+            fonll_recipe[-1]["PTODIS"] = po
+            fonll_recipe[-1]["PTO"] = po - 1
+    return fonll_recipe
+
+
+def produce_fonll_tcards(tcard, tcard_parent_path, theoryid):
+    """Produce the seven fonll tcards from an original tcard and dump them in tcard_parent_path with names from '{theoryid}00.yaml' to '{theoryid}06.yaml'."""
```

Comment:
```suggestion
    """Produce the seven fonll tcards from an original tcard.
    
     We dump them in tcard_parent_path with names from '{theoryid}00.yaml' to '{theoryid}06.yaml'."""
```
There is nothing wrong with using more then one-line docstrings ... the second part is clearly a detail

Reply:
[34fd85d](https://github.com/NNPDF/pineko/pull/99/commits/34fd85d001059a9c3db7a25b7c095b5c5e40eae3)